### DISCUSSION TITLE
Custom host and port for Laravel Echo

### DISCUSSION THREAD

**Discussion by nfragkos** [statamic/cms#5150] [General]
Hi is there a way to provide the host and port option parameters for Laravel Echo?

Example config: 
```
new Echo({
    broadcaster: 'pusher',
    key: process.env.MIX_PUSHER_APP_KEY,
    wsHost: process.env.MIX_PUSHER_HOST,
    wsPort: process.env.MIX_PUSHER_PORT,
    wssPort: process.env.MIX_PUSHER_PORT,
    forceTLS: false,
    encrypted: true,
    disableStats: true,
    enabledTransports: ['ws', 'wss'],
});
```

I want to be able to use Statamic Collboration addon with an open-source pusher server replacement like Soketi and Laravel Websockets but this is not currently possible or not documented anywhere in Statamic's documentation.

**Comment by duncanmcclean** [comment]
I don't think Pusher alternatives are currently supported by the collaboration addon. You'd be best opening a feature request on the collaboration addon repo: https://github.com/statamic/collaboration/issues

**Comment by nfragkos** [comment]
I'm not referring to alternatives but to drop-in replacements. For example Soketi supports the latest Pusher Protocol v7 as mentioned here https://docs.soketi.app/ 

Also Laravel Echo initialisation is happening on Statamic and not on the addon its self: [/resources/js/components/Echo.js#L15](https://github.com/statamic/cms/blob/3.2/resources/js/components/Echo.js#L15)

**Comment by jasonvarga** [comment]
This was discussed in #3927 but was never implemented. Currently we don't have any way to expose those options to you, but we'd be open to a PR.
